packages/pwa-kit-react-sdk/src/ssr/server/react-rendering.js

```diff
@@ -236,6 +237,11 @@ export const render = async (req, res, next) => {
 
     if (includeServerTimingHeader) {
         res.setHeader('Server-Timing', res.__performanceTimer.buildServerTimingHeader())
+
+        // Override cache-control header to no caching when __server_timing is used
+        // This happens after React rendering is complete, ensuring it overrides any
+        // cache headers set by individual page components
+        res.set('Cache-Control', NO_CACHE)
```

Comment:
Bonus: credit to @bendvc for finding this bug with caching html response along with the server timing header. Now we make sure it won't be cached anymore.

FYI you can only see the bug in a remote deployed environment.